index.py

```diff
@@ -76,54 +82,129 @@ def build(self):
+            if i > 0 and term == tokens_list[i-1][0] and curr_tuple[0] == tokens_list[i-1][1][0]:
+                self.dictionary[term].insert(curr_tuple[0], curr_tuple[1], curr_tuple[2], False)
+            else:
+                self.dictionary[term].insert(curr_tuple[0], curr_tuple[1], curr_tuple[2])
+
+            print("done with filling up dictionary term PostingLists")
 
-        # Step 4: Calculate doc_lengths for normalization
+        # Step 5: Calculate doc_lengths for normalization
+        print("Calculating document vector length")
         self.calculate_doc_length()
 
         for _, posting_list in self.dictionary.items():
             posting_list.generate_skip_list()
 
-    # Read the file in and split by the characters '",'
-    # Processes the data set into an array of cases/documents
-    # Note: This function DOES NOT filter punctuation and lower case
+    def get_documents(self):
+        """
+        Returns a list of complete documents which have positional indexes for content, title, court, and date_posted
+        Each complete document is represented by a dictionary with keys: 'doc_id', 'title', 'content', 'date_posted', 'court', 'top_K'
+        and keys for 4 positional_indexes: 'content_positional_indexes', 'title_positional_indexes', 'court_positional_indexes', 'date_posted_positional_indexes'
+        """
+        # Result container for collating all possible dictionary file terms
+        set_of_documents = []
+        documents = self.process_file()
+        print("Done processing file")
+
+        # Then we handle the fields: content, title and court (to generate position index)
+        # of 'content_positional_indexes', 'title_positional_indexes' is made from 'title', 'court_positional_indexes'
+        count = 0
+        for document in documents:
+            document['content_positional_indexes'] = self.generate_positional_indexes(document['content'])  # Part 1: Content
+            document['title_positional_indexes'] = self.generate_positional_indexes(document['title'])  # Part 2: Title
+            document['court_positional_indexes'] = self.generate_positional_indexes(document['court'])  # Part 3: Court
+            document['date_posted_positional_indexes'] = self.generate_positional_indexes(document['date_posted'].split()[0])  # Part 4: Date_posted
+            set_of_documents.append(document)
+
+            # To obtain the top K terms for the current document
+            document['top_K'] = []
+            accumulate_counts = {}
+            self.include_count_contribution_from_pos_ind(accumulate_counts, document['content_positional_indexes'])
+            self.include_count_contribution_from_pos_ind(accumulate_counts, document['title_positional_indexes'])
+            self.include_count_contribution_from_pos_ind(accumulate_counts, document['court_positional_indexes'])
+            self.include_count_contribution_from_pos_ind(accumulate_counts, document['date_posted_positional_indexes'])
```

Comment:
This can be done in the generate_positional_indexes file right, if not we have to go through each document twice

Reply:
It will still need to go through the entire thing again, so put here for clarity and easier to code